software/edge/src/aeris_mesh_agent/aeris_mesh_agent/video_abr_controller.py

```
@@ -0,0 +1,508 @@
+    """Encoder profile application acknowledgement."""
+
+    command_id: int
+    profile_name: str
+    applied_at_ts: float
+    status: str = "ok"
+
+    @staticmethod
+    def from_payload(payload: str | dict[str, object]) -> "EncoderAck":
+        raw = json.loads(payload) if isinstance(payload, str) else payload
+        if not isinstance(raw, dict):
+            raise ValueError("encoder ack payload must be a JSON object")
```

Ruff (0.15.1)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)